Gems/ROS2/Code/Source/FactorySimulation/ConveyorBeltComponentKinematic.h

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/Component/Component.h>
+#include <AzCore/Component/Entity.h>
+#include <AzCore/Component/EntityBus.h>
+#include <AzCore/Component/TickBus.h>
+#include <AzCore/Math/Spline.h>
+#include <AzCore/Math/Transform.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/std/containers/deque.h>
+#include <AzFramework/Physics/Common/PhysicsEvents.h>
+#include <AzFramework/Physics/Common/PhysicsSimulatedBodyEvents.h>
+#include <AzFramework/Physics/Common/PhysicsTypes.h>
+#include <AzFramework/Physics/Material/PhysicsMaterialAsset.h>
+#include <AzFramework/Physics/PhysicsSystem.h>
+#include <ROS2/FactorySimulation/ConveyorBeltRequestBus.h>
+
+namespace ROS2
+{
+    //! Component that simulates a conveyor belt using kinematic physics.
+    //! The conveyor belt is simulated using a spline and number of kinematic rigid bodies.
+    //! The kinematic rigid bodies have their kinematic targets set to interpolate along the spline.
+    //! The component is updating kinematic targets every physic sub-step and creates and despawns rigid bodies as needed.
+    class ConveyorBeltComponentKinematic
```

Comment:
I suggest changing the name so that the Component is the last part, since this is the convention.
I.e. ConveyorBeltComponent.
I think "Kinematic" is unnecessary.
Also, please add an issue to move FactorySimulation directories to separate Gem.